python/core/additions/qgsfeature.py

```diff
@@ -16,12 +16,12 @@
 *                                                                         *
 ***************************************************************************
 """
+from PyQt5.QtCore import QVariant
 
 
 def mapping_feature(feature):
     geom = feature.geometry()
-    fields = [field.name() for field in feature.fields()]
-    properties = dict(list(zip(fields, feature.attributes())))
+    properties = {k: None if (v is None or isinstance(v, QVariant) and v.isNull()) else v for k, v in feature.attributeMap().items()}
```

Comment:
perhaps to make things more explicit:
```suggestion
    properties = {k: None if (v is None or (isinstance(v, QVariant) and v.isNull())) else v for k, v in feature.attributeMap().items()}
```